01-python/outdoors/lake.py

```
@@ -1,3 +1,34 @@
-def draw_lake():
-    print("lake not found")
+import math
+
+
+def draw_lake(area=1, shape="rectangle", width=1, length=1):
+    """
+    Print a lake
+
+    The shape of lake should be square or rectangle.
+    If the lake is square, the area should be a square number.
+    If the lake is rectangle, the given width and length should compatible with the area.
+    """
+    if shape == "square":
+        n = math.sqrt(area)
+        if n % 1 != 0:
+            print("area of square should be square number!")
+            return
+        else:
+            n = int(n)
+        print(" " + n*"-")
+        for i in range(n):
+            print("|" + n*" " + "|")
+        print(" " + n * "-")
+        return
+    if shape == "rectangle":
+        if width*length != area:
+            print("width and length are wrong!")
+        else:
+            print(" " + length * "-")
+            for i in range(width):
+                print("|" + length * " " + "|")
+            print(" " + length * "-")
+            return
+    print("Sorry, now the shape should be square or rectangle.")
```

You should raise an error instead of printing something.

---

and you should raise this error as soon as you can raise it.